core/modules/block/block.module

```diff
@@ -196,6 +196,15 @@ function block_custom_block_form($edit = array(), $stand_alone = TRUE) {
     '#weight' => -20,
     '#id' => 'block-info',
   );
+  $form['description'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Block description'),
+    '#default_value' => $edit['description'],
+    '#maxlength' => 128,
+    '#description' => t('Used in administrative interfaces and not shown to end-users.'),
+    '#weight' => -18,
```

Comment:
The weights of these fields should be:

Admin label: -20
Machine name: -19 <-- very important that this comes after its source, the "Admin label" field. See my comment bellow.
Block description: -18
Display title: -17
Block content: -16

...while at it, can you please change the order of `$form['title'] = array` in the code so that it comes just before `$form['body'] = array(` in order to match the actual field order in the form?

Reply:
Yes, done.